app/screens/edit_profile/edit_profile.tsx

```diff
@@ -0,0 +1,399 @@
+        const unsubscribe = Navigation.events().registerComponentListener({
+            navigationButtonPressed: ({buttonId}: { buttonId: string }) => {
+                switch (buttonId) {
+                    case 'update-profile':
+                        submitUser();
+                        break;
+                    case closeButtonId:
+                        close();
+                        break;
+                }
+            },
+        }, componentId);
+
+        return () => {
+            unsubscribe.remove();
+        };
+    }, [userInfo, isTablet]);
+
+    useEffect(() => {
+        const backHandler = BackHandler.addEventListener('hardwareBackPress', close);
+        return () => {
+            backHandler.remove();
+        };
+    }, []);
+
+    useEffect(() => {
+        if (!isTablet) {
+            setButtons(componentId, {rightButtons: [rightButton!]});
+        }
+    }, [isTablet]);
+
+    const service = currentUser.authService;
+
+    const close = () => {
+        if (isModal) {
+            dismissModal({componentId});
+        } else if (isTablet) {
+            DeviceEventEmitter.emit(Events.ACCOUNT_SELECT_TABLET_VIEW, '');
+        } else {
+            popTopScreen(componentId);
+        }
+
+        return true;
+    };
+
+    const enableSaveButton = (value: boolean) => {
+        const buttons = {
+            rightButtons: [{...rightButton!, enabled: value}],
+        };
+        setCanSave(value);
+        setButtons(componentId, buttons);
+    };
+
+    const submitUser = preventDoubleTap(async () => {
+        enableSaveButton(false);
+        setError(undefined);
+        setUpdating(true);
+        try {
+            const partialUser: Partial<UserProfile> = {
+                email: userInfo.email,
+                first_name: userInfo.firstName,
+                last_name: userInfo.lastName,
+                nickname: userInfo.nickname,
+                position: userInfo.position,
+                username: userInfo.username,
+            };
+
+            const {error: reqError} = await updateMe(serverUrl, partialUser);
+
+            if (reqError) {
+                resetScreen(reqError as Error);
+                return;
+            }
+
+            close();
+        } catch (e) {
+            resetScreen(e as Error);
+        }
+    });
+
+    const resetScreen = (resetError: Error) => {
+        setError(resetError?.message);
+        Keyboard.dismiss();
+        setUpdating(false);
+        enableSaveButton(true);
+        scrollViewRef.current?.scrollToPosition(0, 0, true);
+    };
+
+    const updateField = useCallback((fieldKey: string, name: string) => {
+        const update = {...userInfo};
+        update[fieldKey] = name;
+        setUserInfo(update);
+
+        // @ts-expect-error access object property by string key
+        const currentValue = currentUser[fieldKey];
+        const didChange = currentValue !== name;
+        enableSaveButton(didChange);
+    }, [userInfo]);
```

Comment:
This also seems to depend on currentUser. I guess that currentUser may change in case of a websocket event updating the user.

Reply:
agree

Reply:
Corrected